cmd/gnodev/precompile.go

```diff
@@ -108,15 +125,27 @@ func precompileFile(srcPath string, opts precompileOptions) error {
 	}
 
 	// preprocess.
-	transformed, err := gno.Precompile(string(source), tags, srcPath)
+	f, transformed, err := gno.Precompile(string(source), tags, srcPath)
 	if err != nil {
 		return fmt.Errorf("%w", err)
 	}
 
+	// get imported package path
+	var importPkgPaths []string
+	for _, i := range f.Imports {
+		importPath := i.Path.Value[1 : len(i.Path.Value)-1]
+		prefix := "github.com/gnolang/gno"
+		if strings.HasPrefix(importPath, prefix) {
+			res := strings.TrimPrefix(importPath, prefix)
+			importPkgPaths = append(importPkgPaths, "."+res)
+		}
+	}
+
 	// write .go file.
+	pkgPath := filepath.Dir(srcPath)
 	var targetPath string
 	if opts.Output != defaultPrecompileOptions.Output {
-		targetPath = filepath.Join(opts.Output, targetFilename)
+		targetPath = filepath.Join(opts.Output, pkgPath, targetFilename)
```

Comment:
You can inline `pkgPath` here, as you're not using it anywhere else in the method:

```go
targetPath = filepath.Join(opts.Output, filepath.Dir(srcPath), targetFilename)
```